src/agents/evolution/fitness_functions/StrengthFunction.cc

```diff
@@ -0,0 +1,30 @@
+#include "StrengthFunction.h"
+
+#include "AtomDBSingleton.h"
+#include "Logger.h"
+#include "Utils.h"
+
+using namespace std;
+using namespace atomdb;
+using namespace fitness_functions;
+
+string StrengthFunction::VARIABLE_NAME = "strength";
+
+StrengthFunction::StrengthFunction() { db = AtomDBSingleton::get_instance(); }
+
+float StrengthFunction::eval(shared_ptr<QueryAnswer> query_answer) {
+    float strength = 1.0;
+
+    LOG_INFO("Evaluating strength for " << query_answer->to_string());
+    LOG_INFO("Evaluating strength for " << query_answer->handles.size() << " handles.");
```

Comment:
`handles` is already printed in the INFO message above. If you really want to see the size explicitly you could use a single combined message to print both.